x/mongo/driver/topology/rtt_monitor_test.go

```diff
@@ -265,3 +270,88 @@ func TestMin(t *testing.T) {
 		})
 	}
 }
+
+func TestPercentile(t *testing.T) {
+	cases := []struct {
+		desc       string
+		samples    []time.Duration
+		minSamples int
+		percentile float64
+		want       time.Duration
+	}{
+		{
+			desc:       "Should return 0 for fewer than minSamples samples",
+			samples:    []time.Duration{1, 0, 0, 0},
+			minSamples: 2,
+			percentile: 90.0,
+			want:       0,
+		},
+		{
+			desc:       "Should return 0 for empty samples slice",
+			samples:    []time.Duration{},
+			minSamples: 0,
+			percentile: 90.0,
+			want:       0,
+		},
+		{
+			desc:       "Should return 0 for no valid samples",
+			samples:    []time.Duration{0, 0, 0},
+			minSamples: 0,
+			percentile: 90.0,
+			want:       0,
+		},
+		{
+			desc:       "First tertile when minSamples = 0",
+			samples:    []time.Duration{1, 2, 3, 0, 0, 0},
+			minSamples: 0,
+			percentile: 33.34,
+			want:       1,
+		},
+		{
+			desc: "90th percentile when there are enough samples",
+			samples: []time.Duration{
+				100 * time.Millisecond,
+				200 * time.Millisecond,
+				300 * time.Millisecond,
+				400 * time.Millisecond,
+				500 * time.Millisecond,
+				600 * time.Millisecond,
+				700 * time.Millisecond,
+				800 * time.Millisecond,
+				900 * time.Millisecond,
+				1 * time.Second,
+				0, 0, 0},
+			minSamples: 10,
+			percentile: 90.0,
+			want:       900 * time.Millisecond,
+		},
+		{
+			desc: "10th percentile when there are enough samples",
+			samples: []time.Duration{
+				100 * time.Millisecond,
+				200 * time.Millisecond,
+				300 * time.Millisecond,
+				400 * time.Millisecond,
+				500 * time.Millisecond,
+				600 * time.Millisecond,
+				700 * time.Millisecond,
+				800 * time.Millisecond,
+				900 * time.Millisecond,
+				1 * time.Second,
+				0, 0, 0},
+			minSamples: 10,
+			percentile: 10.0,
+			want:       100 * time.Millisecond,
+		},
+	}
+
+	for _, tc := range cases {
```

Comment:
This is really slick!

Reply:
Not sure exactly which piece you're referring to, but stole most of this structure from @matthewdale 's `TestMin` above haha.

Reply:
It's a very extensible and well thought out pattern!